src/views/domain-page/domain-page-header-info-item/domain-page-header-info-item.types.ts

```diff
@@ -0,0 +1,15 @@
+import type { ReactNode } from 'react';
+
+type LoadingProps = {
+  loading: true;
+  placeholderSize: string;
+};
+
+type LoadedProps = {
+  loading: false;
+  content: string | ReactNode;
+};
+
+export type Props = (LoadingProps | LoadedProps) & {
```

Comment:
The split here is not serving any purpose and adding complexity while using the component. [attached the usage code]
I would expect such a split in the props if the component accepts a complete set of props based on a single prop, which can help the component's user to identify which props are for each type. 
But in the current case loading is just a toggle between two states. Both states will always be sent together and there is no harm in allowing that.

Reply:
```
   <DomainPageHeaderInfoItem
          key={configItem.title}
          title={configItem.title}
          {...(props.loading
            ? {
                loading: true,
                placeholderSize: configItem.placeholderSize,
              }
            : {
                loading: false,
                content: configItem.getContent({
                  domainInfo: props.domainInfo,
                  cluster: props.cluster,
                }),
              })}
        />
```

Reply:
Fair enough. So instead of having LoadingProps and LoadedProps, I can do the following:
```
type Props = {
  loading: boolean;
  content: string | React.ReactNode | undefined;
  placeholderSize: string;
}
```